### PR TITLE
Fix logical apply worker reporting of flush_lsn wrt sync replication.

### PR DESCRIPTION
**Pull request by arssher**
It should take syncrep flush_lsn into account because WAL before it on endpoint restart is lost, which makes replication miss some data if slot had already been advanced too far. This commit adds test reproducing the issue and bumps vendor/postgres to commit with the actual fix.

Related to #1632 and #8097

### CODE REVIEW

File: control_plane/src/endpoint.rs [hunk 1]
```diff
@@ -824,11 +824,12 @@ impl Endpoint {
         // cleanup work to do after postgres stops, like syncing safekeepers,
         // etc.
         //
-        // If destroying, send it SIGTERM before waiting. Sometimes we do *not*
-        // want this cleanup: tests intentionally do stop when majority of
-        // safekeepers is down, so sync-safekeepers would hang otherwise. This
-        // could be a separate flag though.
-        self.wait_for_compute_ctl_to_exit(destroy)?;
+        // If destroying or stop mode is immediate, send it SIGTERM before
+        // waiting. Sometimes we do *not* want this cleanup: tests intentionally
```
Comment (knizhnik) [inline control_plane/src/endpoint.rs:828]: Some tests are using "immediate" shutdown mode to test abnormal compute termination.
I wonder if this change makes this tests less useful.

**Review by knizhnik — CHANGES_REQUESTED** [review]
Why only pg16 is fixed? What about 15/14?

```
XLogRecPtr  sync_rep_flush = WalSndCtl->lsn[SYNC_REP_WAIT_FLUSH];
```
Can WalSndCtl be NULL here?
I think this assignment should be moved inside if body:
```
if ((SyncRepStandbyNames != NULL && SyncRepStandbyNames[0] != '\0'))
{
		XLogRecPtr  sync_rep_flush = WalSndCtl->lsn[SYNC_REP_WAIT_FLUSH];
		local_flush = Min(local_flush, sync_rep_flush);
}
```

**Comment by arssher** [comment]
> Why only pg16 is fixed? What about 15/14?

Just wanted to get approve on one version before proceeding with the others.

File: control_plane/src/endpoint.rs (hunk 1 above)
Comment (arssher) [reply control_plane/src/endpoint.rs:828]: I'd say it should only make them more useful. That is, immediate means as fast as possible, killing compute_ctl is even more 'immediate'.

**Comment by arssher** [comment]
> Can WalSndCtl be NULL here?

I don't think so, it is initialized in WalSndShmemInit.

**Review by knizhnik — APPROVED** [review]

**Pull request merged**